packages/aws-cdk/lib/notices.ts

```diff
@@ -4,104 +4,231 @@ import * as path from 'path';
 import * as fs from 'fs-extra';
 import * as semver from 'semver';
 import { debug, print } from './logging';
+import { Configuration } from './settings';
 import { ConstructTreeNode, loadTreeFromDir, some } from './tree';
 import { flatMap } from './util';
 import { cdkCacheDir } from './util/directories';
 import { versionNumber } from './version';
 
 const CACHE_FILE_PATH = path.join(cdkCacheDir(), 'notices.json');
 
-export interface DisplayNoticesProps {
+export interface NoticesProps {
   /**
-   * The cloud assembly directory. Usually 'cdk.out'.
+   * A `Configuration` instance holding CDK context and settings.
    */
-  readonly outdir: string;
+  readonly configuration: Configuration;
 
   /**
-   * Issue numbers of notices that have been acknowledged by a user
-   * of the current CDK repository. These notices will be skipped.
+   * Include notices that have already been acknowledged.
+   *
+   * @default false
    */
-  readonly acknowledgedIssueNumbers: number[];
+  readonly includeAcknowlegded?: boolean;
+
+}
+
+export interface NoticesPrintOptions {
 
   /**
-   * Whether cached notices should be ignored. Setting this property
-   * to true will force the CLI to download fresh data
+   * Whether to append the total number of unacknowledged notices to the display.
    *
    * @default false
    */
-  readonly ignoreCache?: boolean;
+  readonly showTotal?: boolean;
+
+}
 
+export interface NoticesRefreshOptions {
   /**
-   * Whether to append the number of unacknowledged notices to the display.
+   * Whether to force a cache refresh regardless of expiration time.
    *
    * @default false
    */
-  readonly unacknowledged?: boolean;
-}
+  readonly force?: boolean;
 
-export async function refreshNotices() {
-  const dataSource = dataSourceReference(false);
-  return dataSource.fetch();
+  /**
+   * Data source for fetch notices from.
+   *
+   * @default - WebsiteNoticeDataSource
+   */
+  readonly dataSource?: NoticeDataSource;
 }
 
-export async function displayNotices(props: DisplayNoticesProps) {
-  const dataSource = dataSourceReference(props.ignoreCache ?? false);
-  print(await generateMessage(dataSource, props));
-  return 0;
-}
+export class NoticesFormatter {
```

Comment:
We don't need to export this class.

Reply:
I'm actually using this in tests...not great I know, i'll see what I can do but I think its still ok to ship (previous code also exported internal functions)

Reply:
I'm going to keep it. It helps make the tests clean and just not worth the refactor right now.